### PR TITLE
IRGen: Emit the protocol requirement descriptor async flag for initializers

### PR DESCRIPTION
**Pull request by aschwaighofer**
rdar://88180938

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci test

File: lib/IRGen/GenMeta.cpp [hunk 1]
```diff
@@ -240,8 +240,12 @@ static bool needsForeignMetadataCompletionFunction(IRGenModule &IGM,
 
 template <class Flags>
 static Flags getMethodDescriptorFlags(ValueDecl *fn) {
-  if (isa<ConstructorDecl>(fn))
-    return Flags(Flags::Kind::Init); // 'init' is considered static
+  if (isa<ConstructorDecl>(fn)) {
+    auto flags = Flags(Flags::Kind::Init); // 'init' is considered static
+    if (auto *afd = dyn_cast<AbstractFunctionDecl>(fn))
```
Comment (nate-chandler) [inline lib/IRGen/GenMeta.cpp:245]: Since fn isa ConstructorDecl could, this be a plain cast?

**Review by nate-chandler — APPROVED** [review]
LGTM!

**Comment by nate-chandler** [comment]
@swift-ci please test macos platform

**Pull request merged**